modules/tinymce/src/core/main/ts/api/EditorObservable.ts

```diff
@@ -54,12 +54,13 @@ const getEventTarget = (editor: Editor, eventName: string): Node => {
   return editor.getBody();
 };
 
-const isListening = (editor: Editor) => !editor.hidden && !isReadOnly(editor);
+const isListening = (editor: Editor, e: Event) => !editor.hidden && (!isReadOnly(editor) || isReadOnlyAllowedEvent(e));
```

Comment:
This is confusing/mixing the logic here, as. it's meant to be divided into listening to any event and then processing when in readonly mode. As such, this would be better handled by emitting the event within the `processReadonlyEvents` function instead if it's in the whitelist, as its very purpose is to decide what to do with events while in readonly mode.

- [ ] Revert these changes and move this logic into `processReadonlyEvents` please

Reply:
done